gbpservice/neutron/services/servicechain/plugins/ncp/node_drivers/nfp_node_driver.py

```diff
@@ -641,9 +682,65 @@ def _is_node_order_in_spec_supported(self, context):
             raise InvalidNodeOrderInChain(
                     node_order=allowed_chain_combinations)
 
+    def _get_consumers_for_provider(self, context, provider):
+        '''
+        {
+            consuming_ptgs_details: [{'ptg': <>, 'subnets': <>}]
+            consuming_eps_details: []
+        }
+        '''
+
+        consuming_ptgs_details = []
+        consuming_eps_details = []
+
+        provided_prs_id = provider['provided_policy_rule_sets'][0]
+        provided_prs = context.gbp_plugin.get_policy_rule_set(
+            context.plugin_context, provided_prs_id)
+        consuming_ptg_ids = provided_prs['consuming_policy_target_groups']
+        consuming_ep_ids = provided_prs['consuming_external_policies']
+    
+        consuming_ptgs = context.gbp_plugin.get_policy_target_groups(
+                context.plugin_context, filters={'id':consuming_ptg_ids})
+        consuming_eps_details = context.gbp_plugin.get_external_policies(
+                context.plugin_context, filters={'id': consuming_ep_ids})
```

Comment:
If no external policies available, it'll return empty list, no need to define that above.